validators.py

```diff
@@ -1,7 +1,23 @@
-from pydantic import BaseModel, conlist, conint, Field
+from datetime import datetime
+
+from pydantic import BaseModel, validator
 
 
 class DatesAndPriceCheckInputValidator(BaseModel):
-    listing_ids: conlist(conint(gt=0), min_items=1) = Field(
-        title="Listing ids to check on"
-    )
+    listing_ids: list
+    desired_checkin: str
```

Comment:
```suggestion
    desired_checkin: Optional[date]
```
By doing so, Pydantic will convert the user input `str` type object into a `datetime.date` type.

You will have to do the following imports
```python
from datetime import datetime, date
from typing import Optional
```

Reply:
Noted. Will make the change